src/ui/veles_mainwindow.cc

```diff
@@ -335,8 +344,15 @@ bool TabBarEventFilter::eventFilter(QObject *watched, QEvent *event) {
 
 bool TabBarEventFilter::mouseMove(QTabBar* tab_bar, QMouseEvent* event) {
   if(dragged_tab_bar_) {
-    if ((event->pos() - drag_init_pos_).manhattanLength()
-        > QApplication::startDragDistance()) {
+    bool horizontal_tabs =
+        dragged_tab_bar_->shape() == QTabBar::RoundedNorth
+        || dragged_tab_bar_->shape() == QTabBar::RoundedSouth
+        || dragged_tab_bar_->shape() == QTabBar::TriangularNorth
+        || dragged_tab_bar_->shape() == QTabBar::TriangularSouth;
+
+    if ((horizontal_tabs ? (event->pos() - drag_init_pos_).y()
+        : (event->pos() - drag_init_pos_).x())
+        > 5 * QApplication::startDragDistance()) {
```

Comment:
I think this value (5 * startDragDistance) may require tuning later, so would be nice to have it on a const, rather than hardcode it here.